blocks/math.js

```diff
@@ -82,12 +82,19 @@ Blockly.Blocks['math_positive_number'] = {
    * @this Blockly.Block
    */
   init: function() {
-    this.setHelpUrl(Blockly.Msg.MATH_NUMBER_HELPURL);
-    this.setColour(Blockly.Blocks.math.HUE);
-    this.appendDummyInput()
-        .appendField(new Blockly.FieldNumber('0', 0, Infinity), 'NUM');
-    this.setOutput(true, 'Number');
-    this.setOutputShape(Blockly.OUTPUT_SHAPE_ROUND);
+    this.jsonInit({
+      "message0": "%1",
+      "args0": [
+        {
+          "type": "field_number",
+          "name": "NUM",
+          "min": 0
```

Comment:
I'm not entirely sure I'm understanding correctly, but my interpretation of this conversation is that you want to know whether decimal differences in pen color affect the color being displayed.

If so, the answer is yes:
![colordiff](https://cloud.githubusercontent.com/assets/5458180/16714716/b021136c-4699-11e6-8300-a27919395e76.png)
That being said, I'm not sure backwards compatibility is really a big deal in this case. If pen colors were rounded to the nearest integer, I find it hard to believe anybody would mind.